DataPipe naming convension update

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#51262 DataPipe naming convension update**

This PR is to change the naming convention of what previously called 'dataset'. Instead, we named them DataPipe. 
This PR is specifically for ListDirFilesIterableDataset and LoadFilesFromDiskIterableDataset.

And we provide the following way to import them.
1. partial import as a `datapipes` module
```
import torch.utils.data.datapipes as dp
dp.iter.ListDirFiles
dp.iter.LoadFilesFromDisk
```
2. direct import the DataPipe class
```
from torch.utils.data.datapipes import ListDirFiles, LoadFilesFromDisk
```

This PR also added the support for recursively scanning the folders.

Next step will be Tar/Zip/Gz dataset -> datapipe

Differential Revision: [D26120628](https://our.internmc.facebook.com/intern/diff/D26120628)

Commit:
DataPipe naming convension update

[ghstack-poisoned]

ejguan commented on torch/utils/data/datapipes/__init__.py:
```diff
@@ -0,0 +1 @@
+from torch.utils.data.datapipes.iter import *
```
Just considering future map datapipe with same name, we should keep the API only within `iter`.

ejguan reviewed (approved):
LGTM and one comment below.

Commit:
Update on "DataPipe naming convension update"


This PR is to change the naming convention of what previously called 'dataset'. Instead, we named them DataPipe. 
This PR is specifically for ListDirFilesIterableDataset and LoadFilesFromDiskIterableDataset.

And we provide the following way to import them.
1. partial import as a `datapipes` module
```
import torch.utils.data.datapipes as dp
dp.iter.ListDirFiles
dp.iter.LoadFilesFromDisk
```
2. direct import the DataPipe class
```
from torch.utils.data.datapipes import ListDirFiles, LoadFilesFromDisk
```

This PR also added the support for recursively scanning the folders.

Next step will be Tar/Zip/Gz dataset -> datapipe

Differential Revision: [D26120628](https://our.internmc.facebook.com/intern/diff/D26120628)

[ghstack-poisoned]

Commit:
Update on "DataPipe naming convension update"


This PR is to change the naming convention of what previously called 'dataset'. Instead, we named them DataPipe. 
This PR is specifically for ListDirFilesIterableDataset and LoadFilesFromDiskIterableDataset.

And we provide the following way to import them.
1. partial import as a `datapipes` module
```
import torch.utils.data.datapipes as dp
dp.iter.ListDirFiles
dp.iter.LoadFilesFromDisk
```
2. direct import the DataPipe class
```
from torch.utils.data.datapipes import ListDirFiles, LoadFilesFromDisk
```

This PR also added the support for recursively scanning the folders.

Next step will be Tar/Zip/Gz dataset -> datapipe

Differential Revision: [D26120628](https://our.internmc.facebook.com/intern/diff/D26120628)

[ghstack-poisoned]
